utils.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+const isObject = (item) => {
```

Comment:
An easy way to do reliable type-checking is the following, which I prefer to use:

```javascript
const checkIsType = (value, type) =>
    Object.prototype.toString.call(value)
        .substring(8, -1)
        .toLowerCase() === type.toLowerCase();
```

And is called with `checkIsType(myValue, 'object');` See [Todd Motto's post on using the method](https://toddmotto.com/understanding-javascript-types-and-reliable-type-checking/). This also gives us additional type checking if we need it moving forward.